### ISSUE TITLE
Filter AnyStr with strings False Positive

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1683]
**Describe the bug**
I expect filter to work when given os.path.exists and list of strings. The type for os.path.exists is Union[AnyStr, _PathLike[AnyStr]] -> bool and AnyStr includes str as one possibility. 

**To Reproduce**
Run pyright on this code sample.

```
import os
result = filter(os.path.exists, ["hello", "world"])
```

The result for me is,

```
 error: No overloads for "filter" match the provided arguments
    Argument types: ((path: AnyStr@exists | _PathLike[AnyStr@exists]) -> bool, list[str]) (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
```

**Expected behavior**
No errors reported. I tested this with mypy and it passes fine.

**VS Code extension or command-line**
I'm running pyright from the command line. The version is 1.1.124.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.125, which I just published. It will also be included in the next release of pylance.
